modules/age.nix

```diff
@@ -16,13 +16,15 @@ let
 
   identities = builtins.concatStringsSep " " (map (path: "-i ${path}") cfg.sshKeyPaths);
   installSecret = secretType: ''
-    echo "decrypting ${secretType.file} to ${secretType.path}..."
-    TMP_FILE="${secretType.path}.tmp"
-    mkdir -p $(dirname ${secretType.path})
+    echo "decrypting '${secretType.file}' to '${cfg.secretsMountPoint}/$_count/${secretType.name}'..."
+    TMP_FILE="${cfg.secretsMountPoint}/$_count/${secretType.name}.tmp"
+    mkdir -p "$(dirname "${cfg.secretsMountPoint}/$_count/${secretType.name}")"
+    mkdir -p "$(dirname "${secretType.path}")"
     (umask 0400; LANG=${config.i18n.defaultLocale} ${ageBin} --decrypt ${identities} -o "$TMP_FILE" "${secretType.file}")
     chmod ${secretType.mode} "$TMP_FILE"
     chown ${secretType.owner}:${secretType.group} "$TMP_FILE"
-    mv -f "$TMP_FILE" '${secretType.path}'
+    mv -f "$TMP_FILE" "${cfg.secretsMountPoint}/$_count/${secretType.name}"
+    [ "${secretType.path}" != "/run/secrets/${secretType.name}" ] && ln -sfn "/run/secrets/${secretType.name}" "${secretType.path}"
```

Comment:
So IIUC, if `secretType.path` is e.g. `/foo/bar`, the secret will be symlinked both into `/foo/bar` and `/run/secrets/bar`.

Is this necessarily what a user wants?

Reply:
I'd say yes (and it's how sops-nix behaves, IIRC). `/run/secrets` should be the single source of truth for the secrets on the system, so that switching a secrets generation "expires" any removed secrets (resulting in a dead symlink, if names don't collide).

This would definitely increase complexity, but maybe we could store a sort of manifest in `/run/secrets.d` that lists the secret name and its `secretType.path`, so we can `unlink` them all prior to switching secrets generations.